tsdb/chunks/head_chunks.go

```diff
@@ -273,6 +273,25 @@ func NewChunkDiskMapper(reg prometheus.Registerer, dir string, pool chunkenc.Poo
 	return m, m.openMMapFiles()
 }
 
+// Chunk encodings for out-of-order chunks.
+// These encodings must be only used by the Head block for its internal bookkeeping.
+const (
+	OutOfOrderMask        = 0b10000000
+	RestoreOutOfOrderMask = 0b01111111
+)
+
+func (cdm *ChunkDiskMapper) OutOfOrderEncoding(sourceEncoding chunkenc.Encoding) chunkenc.Encoding {
+	return sourceEncoding | OutOfOrderMask
+}
+
+func (cdm *ChunkDiskMapper) IsOutOfOrderChunk(e chunkenc.Encoding) bool {
+	return (e & OutOfOrderMask) != 0
+}
+
+func (cdm *ChunkDiskMapper) OutOfOrderDecoding(sourceEncoding chunkenc.Encoding) chunkenc.Encoding {
```

Comment:
Same as above, we're applying a mask to decode

Reply:
Ok, will rename it as suggested.

Reply:
After getting rid of `RestoreOutOfOrderMask` as suggested, this can be renamed as `UnapplyOutOfOrderMask`.

Reply:
Ok, will do